src/ResourceManager/Automation/Commands.Automation/Common/IAutomationClient.cs

```diff
@@ -291,8 +291,8 @@ public interface IAutomationClient
         IEnumerable<JobSchedule> ListJobSchedulesByRunbookName(string resourceGroupName, string automationAccountName, string runbookName);
 
         IEnumerable<JobSchedule> ListJobSchedulesByScheduleName(string resourceGroupName, string automationAccountName, string scheduleName);
-
-        JobSchedule RegisterScheduledRunbook(string resourceGroupName, string automationAccountName, string runbookName, string scheduleName, IDictionary parameters);
+                
+        JobSchedule RegisterScheduledRunbook(string resourceGroupName, string automationAccountName, string runbookName, string scheduleName, IDictionary parameters, string RunOn);
```

Comment:
RunOn Parameter is not mandatory, but I see the Public Function RegisterScheduleRunBook signature has changed and now requires users to provide RunOn parameter.
Now that being said, can your users call RegisterScheduleRunBook from Automation client, if yes then this looks like a breaking change.
Correct me if I am wrong. Won't an overload will be the right approach.